extensions/amazon-search/src/index.tsx

```diff
@@ -49,24 +52,64 @@ export default function Command() {
     keepPreviousData: true,
   });
 
-  const items = data ? data.suggestions.map((suggestion) => suggestion.value) : [];
+  const handleSearch = (text: string) => {
+    setSearchText(text);
+    if (text && !recentSearches.includes(text)) {
+      const updatedSearches = [text, ...recentSearches].slice(0, MAX_RECENT_SEARCHES);
+      setRecentSearches(updatedSearches);
+    }
+  };
+
+  const handleRemoveSearchItem = (query: string) => {
+    setRecentSearches(recentSearches.filter((item) => item !== query));
+  };
+
+  const handleClearSearchHistory = () => {
+    setRecentSearches([]);
+  };
+
+  const suggestions = data ? data.suggestions.map((suggestion) => suggestion.value) : [];
 
   return (
-    <List isLoading={isLoading} searchBarPlaceholder="Search Amazon..." onSearchTextChange={setSearchText} throttle>
-      <List.Section title="Suggestions" subtitle={`${items.length}`}>
-        {items.map((item, index) => (
-          <List.Item
-            key={index}
-            title={item}
-            actions={
-              <ActionPanel>
-                <Action.OpenInBrowser url={`https://www.amazon.${tld}/s?k=${encodeURIComponent(item)}`} />
-              </ActionPanel>
-            }
-          />
-        ))}
-      </List.Section>
-      <List.EmptyView icon="amazon-emptyview.png" title="No Results" />
+    <List isLoading={isLoading} searchBarPlaceholder="Search Amazon..." onSearchTextChange={handleSearch} throttle>
+      {searchText.length > 0 && (
+        <List.Section title="Suggestions" subtitle={`${suggestions.length}`}>
+          {suggestions.map((item, index) => (
+            <List.Item
+              key={index}
+              title={item}
+              actions={
+                <ActionPanel>
+                  <Action.OpenInBrowser url={`https://www.amazon.${tld}/s?k=${encodeURIComponent(item)}`} />
+                </ActionPanel>
+              }
+            />
+          ))}
+        </List.Section>
+      )}
+
+      {searchText.length === 0 && (
+        <List.Section title="Recent Searches">
+          {recentSearches.map((item, index) => (
+            <List.Item
+              key={index}
+              title={item}
+              actions={
+                <ActionPanel>
+                  <Action.OpenInBrowser url={`https://www.amazon.${tld}/s?k=${encodeURIComponent(item)}`} />
+                  <Action title="Remove Search Item" onAction={() => handleRemoveSearchItem(item)} />
+                  <Action title="Clear Search History" onAction={handleClearSearchHistory} />
```

Comment:
Maybe add `style={Action.Style.Destructive}` and confirmAlert?

Reply:
Done.